components/mediation-initializer/org.wso2.carbon.mediation.transport.handlers/src/main/java/org/wso2/carbon/mediation/transport/handlers/requestprocessors/swagger/format/SwaggerYamlProcessor.java

```diff
@@ -47,9 +62,119 @@ public void process(CarbonHttpRequest request, CarbonHttpResponse response,
         if (api == null) {
             handleException(request.getRequestURI());
         } else {
+            //Retrieve from registry
+            String responseString;
             Yaml yamlDefinition = new Yaml();
-            String responseString = yamlDefinition.dumpAsMap(new GenericApiObjectDefinition(api).getDefinitionMap());
+
+            try {
+                String defFromRegistry = retrieveFromRegistry(api, request);
+                if (defFromRegistry != null) {
+                    JsonParser jsonParser = new JsonParser();
+                    responseString = yamlDefinition.dumpAsMap(jsonObjectToMap(jsonParser.parse(defFromRegistry)));
+                } else {
+                    responseString = yamlDefinition.dumpAsMap(new GenericApiObjectDefinition(api).getDefinitionMap());
+                }
+            } catch (RegistryException e) {
+                String errMsg = "Error occurred while retrieving swagger definition from registry";
+                log.error(errMsg, e);
+                throw new AxisFault(errMsg, e);
+            }
+
+
             updateResponse(response, responseString, SwaggerConstants.CONTENT_TYPE_YAML);
         }
     }
+
+    /**
+     * Function to convert from GSON object model to generic map model.
+     *
+     * @param jsonElement gson object to transform
+     * @return
+     */
+    private Map jsonObjectToMap(JsonElement jsonElement) {
```

Comment:
It is better if we move utility functions to a separate class.

Reply:
Fixed, Thanks